### ISSUE TITLE
Overlapping loops to create infinite cross-fade transitions

### ISSUE THREAD

**Issue by docherty** [juliangarnier/anime#194]
I'm trying to create a very simple animation and I nearly have it working but I'm stuck trying to get the end-->beginning of the loop to work correctly.

I have 3 divs which I want to fade up and then down again (opacity 0 --> 1-->0) sequentially. Importantly, I want them to "cross fade" i.e. when one starts to fade down the next in the sequence starts to fade up. This is simple for the transitions between first-->second, second-->third but I can't figure out how to get third-->first to avoid a blank moment as the timeline resets after third.

My timeline logic is this:
0->1s change div[0] opacity = 1
3s->4s change div[0] opacity = 0
3s ->4s change div[1] opacity = 1
6s->7s change div[1] opacity = 0
6s->7s change div[2] opacity = 1
9s->10s change div[2] opacity = 0
9s->10s change div[0] opacity = 1 __// this isn't possible?!__
...
The HTML snippet:
```
    <div class="pressLogo"><img src="img/logo1.svg"></div>
    <div class="pressLogo"><img src="img/logo2.svg"></div>
    <div class="pressLogo"><img src="img/logo3.svg"></div>
```
The js:
```
    var animatePressLogosTL = anime.timeline({
      loop: true
    })

    animatePressLogosTL.add({
      targets: '.pressLogo',
      opacity: [{value: 0, duration: 1000},{value: 1, duration: 2000},{value: 0, duration: 1000}],
      easing: 'easeOutQuad',
      delay: function (el, i) {
        return i * 3000
      }
    })
```
Is there a function of anime.js that would help solve this? If not, perhaps I could duplicate the first element at the end and control the timeline to jump to a frame after the start? Anyone got any experience of this sort of thing? Thanks.

**Comment by Splendorr** [comment]
@docherty I'm trying to accomplish something along these lines. I want an infinite loop that overlaps with itself somehow. I have a series of boxes translating into the distance, but i want a steady stream of them without a gap as the animation finishes and starts again. I tried seeking around the timeline, but couldn't get that working. I got something working with setTimeouts on separate animations, but that creates other problems. 

Did you ever figure out a way to do this?

(My project is confidential right now, and I haven't worked up abstracted code yet.)

**Comment by Splendorr** [comment]
Right now I'm thinking the solution for me will be to create and append elements like in this example: https://codepen.io/juliangarnier/pen/PpoboR

**Comment by docherty** [comment]
@Splendorr my use case was pretty simple so I just added a 4th element to the end of the sequence (the same as the first element). Then I chopped the "loop" in a place where I could simply cut from element 4 to element 1 without it being noticeable (i.e. both are opacity 1). If I wanted all the elements on screen for the same amount of time I made sure that the total duration of 1 + 4 = the right length. If I need to fade the sequence up at the beginning I put the whole thing in a wrapper and fade that up. Hacky but it works!
